File: src/data/AttendanceDAO.py
# data/attendance_dao.py
from config.db_connection import DBConnection

class AttendanceDAO:
    def __init__(self):
        self.conn = DBConnection().get_connection()

    def save_attendance(self, student_id, status):
        cursor = self.conn.cursor()
        cursor.execute(
            "INSERT INTO attendance (student_id, status) VALUES (?, ?)",
            (student_id, status)
        )
        self.conn.commit()

    def get_all_attendance(self):
        cursor = self.conn.cursor()
        cursor.execute("SELECT * FROM attendance")
        return cursor.fetchall()

File: src/data/AttendanceDAO.java
package data;

/**
 * Data Access Object (DAO)
 * Handles database operations
 */
public class AttendanceDAO {

    public void saveAttendance(String studentId, String date, String status) {
        // Simulated database operation
        System.out.println(
            "Attendance Saved -> Student ID: " + studentId +
            ", Date: " + date +
            ", Status: " + status
        );
    }

    public void getAttendanceByStudent(String studentId) {
        // Simulated fetch operation
        System.out.println(
            "Fetching attendance records for Student ID: " + studentId
        );
    }
}

